test/server/serverTest.js

```diff
@@ -1,8 +1,97 @@
 'use strict';
 
-const assert = require('assert');
+const assert = require('assert').strict;
+const http = require('http');
+const fs = require('fs');
+const createServer = require('../../src/testServer/testServer.js');
 
-assert.ok(true, 'Must be true');
+const url = 'http://localhost:8000/';
+const paths = [ 'profile/', 'cards/', 'transactions/', 'profile' ];
 
-// TODO: Create some files hierarchy in this directory, test whatever can be tested.
-// Tests are run by 'npm test' command (see package.json).
+const testCase = (credentials, code, expected) => (
+  { credentials, code, expected }
+);
+const tests = [
+  testCase('alegator:good_job', 200, ''),
+  testCase('Anauthorized', 401, 'You should authorize to access the site'),
+  testCase(':', 401, 'Login and password should be specified'),
+  testCase('alegator:', 401, 'Login and password should be specified'),
+  testCase(':good_job', 401, 'Login and password should be specified'),
+  testCase('saym:onloh', 403, 'Incorrect login or password :('),
+  testCase('alegator:bad_job', 403, 'Incorrect login or password :('),
+];
+
+const getOptionsWithAuthorization = credentials => {
+  const credentialsBase64 = Buffer.from(credentials).toString('base64');
+  let options = { 
+      headers: { 
+        Authorization: `Basic ${ credentialsBase64 }` 
+      }
+  };
+  return options;
+};
+
+const checkStatusCode = (statusCode, expectedCode, path, credentials) => {
+  assert.strictEqual(
+    statusCode, expectedCode,
+    `Wrong status code, expected ${expectedCode}, got ${statusCode}
+    Path: ${path}
+    Credentials: ${credentials}`
+  );
+};
+
+const checkMsg = (msg, expected, path, credentials) => {
+  assert.strictEqual(
+    JSON.parse(msg), `&#127819${expected}`, 
+    `Wrong error msg, expected &#127819${expected}, got ${msg}
+    Path: ${path}
+    Credentials: ${credentials}`
+  );
+};
+
+const logSuccess = (path, credentials, statusCode, msg) => {
+  console.log(
+    `Test successful: ${path} ${credentials}, code: ${statusCode},
+    got msg: ${msg.toString()}\n`
+  );
+};
+
+const testPath = path => {
+  const fullPath = url + path;
+  for (const test of tests) {
+    const { credentials, code, expected } = test;
+    let options = (credentials === 'Anauthorized') ?
```

Comment:
Unauthorized*

Reply:
It would be better if you do not check for non-authorization by comparing strings. Because if we want to test this case with other wrong credentials, the code logic will be wrong. It would be better if you check the credentials in getOptionsWithAuthorization and return an empty object from it.